e2e/fixtures/client/main.go

```diff
@@ -21,8 +21,14 @@ func main() {
 		PreferGo: true,
 		Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
 			d := net.Dialer{Timeout: time.Second}
-			ns := nameservers[rand.Intn(len(nameservers)-1)]
-			ns = ns[:len(ns)-1] // remove trailing period added for some reason by azure dns
+			var ns string
+			if len(nameservers) > 1 {
+				ns = nameservers[rand.Intn(len(nameservers)-1)]
+				ns = ns[:len(ns)-1] // remove trailing period added for some reason by azure dns
+			} else {
+				ns = nameservers[0]
```

Comment:
why do we not remove the trailing period in the else as well? we should probably do it after the if else

Reply:
it has to do with the kind of value we're feeding in. if multiple nameservers are fed in, we know that's coming from a public DNS zone's NS server info, and the addresses for those contain a dot at the end.

If there's only one address fed in, we know it's referring to the NS server of a private cluster, and so there's no trailing dot we'd need to remove.

Reply:
documenting that with a comment would be good

Reply:
will do!

Reply:
FWIW, trailing periods at the end of nameservers is documented in [RFC1034](https://www.rfc-editor.org/rfc/rfc1034). It isn't a quirky Azure DNS behavior.